### PR TITLE
Reproducibility script for time-series search

### PR DESCRIPTION
**Pull request by t-sakai-kure**
Hi, I added scripts for reproducing time-series search experiments in TSPulse paper.
README: https://github.com/ibm-granite/granite-tsfm/blob/search_repro/notebooks/hfdemo/tspulse/search/README.md

### CODE REVIEW

File: notebooks/hfdemo/tspulse/search/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import argparse
+
+import faiss
+import numpy as np
+import pandas as pd
+import torch
+from datautils import (
```
[the remark this answers is not in the document]
Comment (wgifford) [reply notebooks/hfdemo/tspulse/search/benchmark.py:7]: Can we load it with something in the public repo?

[the remark this answers is not in the document]
Comment (t-sakai-kure) [reply notebooks/hfdemo/tspulse/search/benchmark.py:7]: In `ucr_preprocessing.py`, each dataset in UCR is loaded using `time_series_classification_preprocessor` and `ClassificationDFDataset`. Then, resampling is performed so that the dataset in UCR contains 10 samples. Datasets with fewer than 10 samples for a certain class are excluded from processing. After that, the `UCRDataset` class creates labels for the search task. If resampling and label creation can be done using something in the public repo, it may be possible to execute the entire process without implementing the current class and script.

[the remark this answers is not in the document]
Comment (t-sakai-kure) [reply notebooks/hfdemo/tspulse/search/benchmark.py:7]: @wgifford At this point, it seems difficult to replace the class/function in the public repo. Since the `UCRDataset` class is responsible for generating two types of labels for this experiment, we need to implement a wrapper class specifically for handling them.

[the remark this answers is not in the document]
Comment (wgifford) [reply notebooks/hfdemo/tspulse/search/benchmark.py:7]: For now it’s ok, but if we need these functionalities for some use case we should integrate them in the core library.
